### ISSUE TITLE
Initialization takes time

### ISSUE THREAD

**Issue by rbardou** [mirage/mirage-crypto#43]
If you make an executable which links with just `mirage-crypto-pk`, this executable takes a significant time to load (more than 1s on my device).

This is with `mirage-crypto-pk` version 0.6.1.

Here are some steps to reproduce:
```
[Thu 17:47:35] (romain@tarteflambee) /tmp/test> ls
dune  main.ml
[Thu 17:48:28] (romain@tarteflambee) /tmp/test> cat dune
(executables
 (names main)
 (libraries mirage-crypto-pk)
 (flags (:standard -linkall)))
[Thu 17:48:30] (romain@tarteflambee) /tmp/test> cat main.ml
let () = print_endline "LOADED"
[Thu 17:48:32] (romain@tarteflambee) /tmp/test> dune build main.exe
Info: Creating file dune-project with this contents:
| (lang dune 2.4)
[Thu 17:48:48] (romain@tarteflambee) /tmp/test> time _build/default/main.exe
LOADED
_build/default/main.exe  1.01s user 0.00s system 99% cpu 1.006 total
```

**Comment by hannesm** [comment]
would you mind to try #42, and see if this helps?

**Comment by rbardou** [comment]
With #42 the problem seems to be solved :)

**Comment by hannesm** [comment]
fixed in #42, released as 0.6.2 (https://github.com/ocaml/opam-repository/pull/16045)
